apps/openmw/mwmechanics/aiwander.cpp

```diff
@@ -391,18 +393,30 @@ namespace MWMechanics
         short unsigned& idleAnimation = storage.mIdleAnimation;
         idleAnimation = getRandomIdle();
 
+        // If we should be moving
         if (!idleAnimation && mDistance)
         {
             storage.mState = Wander_MoveNow;
+            return;
         }
-        else
+        // If we aren't going to just stand
+        if(idleAnimation)
         {
-            // Play idle animation and recreate vanilla (broken?) behavior of resetting start time of AIWander:
-            MWWorld::TimeStamp currentTime = MWBase::Environment::get().getWorld()->getTimeStamp();
-            mStartTime = currentTime;
-            playIdle(actor, idleAnimation);
-            storage.mState = Wander_IdleNow;
+            // If the idle animation actually exists
+            if(std::find(storage.mBadIdles.begin(), storage.mBadIdles.end(), idleAnimation)==storage.mBadIdles.end())
+            {
+                if(!playIdle(actor, idleAnimation))
+                {
+                    storage.mBadIdles.push_back(idleAnimation);
+                    storage.mState = Wander_ChooseAction;
+                    return;
+                }
+            }
         }
+        // Recreate vanilla (broken?) behavior of resetting start time of AIWander:
+        mStartTime = MWBase::Environment::get().getWorld()->getTimeStamp();
+        storage.mState = Wander_IdleNow;
+        return;
```

Comment:
Yep, but I wanted consistency with all the other "storage.mState = ..." commands.
My natural development style assumes that I'm an idiot.